src/Core/Content/Cookie/SalesChannel/CookieRoute.php

```diff
@@ -35,7 +41,31 @@ public function getDecorated(): AbstractCookieRoute
     public function getCookieGroups(Request $request, SalesChannelContext $salesChannelContext): CookieRouteResponse
     {
         $cookieGroups = $this->cookieProvider->getCookieGroups($salesChannelContext);
+        $hash = $this->generateCookieConfigurationHash(clone $cookieGroups);
 
-        return new CookieRouteResponse($cookieGroups);
+        return new CookieRouteResponse($cookieGroups, $hash);
+    }
+
+    private function generateCookieConfigurationHash(CookieGroupCollection $cookieGroups): string
```

Comment:
should we make this more explict by manually assembling the hash instead of encoding the entire object. I fear that this could break unexpectedly because something new added to the struct or plugin added something.